<?xml version="1.0"?>
<launch>
  <!-- The vehicle types available for the vehicle_type parameter are:
       POLARIS_GEM: The Polaris GEM e2, e4, e6, and eLXD.
       POLARIS_RANGER: The Polaris Ranger series of off-road vehicles.
       INTERNATIONAL_PROSTAR_122: The International PROSTAR+ 122 semi-tractor trailer.
       LEXUS_RX_450H: The hybrid version of the Lexus RX-450 SUV. -->
  <arg name="pacmod_vehicle_type" default="POLARIS_GEM" />
  <arg name="use_kvaser" default="false" />
  <arg name="kvaser_hardware_id" default="27733" />
  <arg name="kvaser_circuit_id" default="0" />
  <arg name="use_socketcan" default="false" />
  <arg name="socketcan_device" default="can0" />
  <arg name="namespace" default="pacmod" />

  <group ns="$(arg namespace)">
    <include file="$(find kvaser_interface)/launch/kvaser_can_bridge.launch" if="$(arg use_kvaser)">
      <arg name="can_hardware_id" value="$(arg kvaser_hardware_id)" />
      <arg name="can_circuit_id" value="$(arg kvaser_circuit_id)" />
      <arg name="can_bit_rate" value="500000" />
    </include>

    <node pkg="socketcan_bridge" type="socketcan_bridge_node" name="socketcan_bridge" if="$(arg use_socketcan)">
      <param name="can_device" value="$(arg socketcan_device)" />
      <remap from="received_messages" to="can_tx" />
      <remap from="sent_messages" to="can_rx" />
    </node>

    <node pkg="pacmod2" type="pacmod2" name="pacmod2">
      <param name="vehicle_type" value="$(arg pacmod_vehicle_type)" />
    </node>
  </group>
</launch>
